### PR TITLE
election, member: adjust some election methods

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Adjust some election methods.

### What is changed and how it works?

* Make `member.Leadership` non-public.
* Make `WatchLeader` a more general method because the TSO allocator will also use the same logic to watch the allocator changing.
* Adjust some functions' signatures to make them more sense in using.
* Add and update some comments.

### Check List

Tests

- Unit test
- Integration test

### Release note

* Adjust some election methods.

### CODE REVIEW

File: server/member/member.go [hunk 1]
```diff
@@ -115,21 +112,53 @@ func (m *Member) GetLeader() *pdpb.Member {
...
+func (m *Member) KeepLeadership(ctx context.Context) {
+	m.leadership.Keep(ctx)
+}
+
+// CheckLeadership is used to check whether a leadership of
+// PD member is still avalibale.
+func (m *Member) CheckLeadership() bool {
+	return m.leadership.Check()
+}
+
```
Comment (Yisaer) [inline server/member/member.go:161]: I think we can hide the `LeaderShip` in these methods here.

Comment (JmPotato) [reply server/member/member.go]: Done.

**Review by Yisaer — APPROVED** [review]
LGTM

**Pull request merged**